### ISSUE TITLE
Error: Asset not found

### ISSUE THREAD

**Issue by mrnetforge** [reggie3/react-native-webview-quilljs#32]
![image from ios](https://user-images.githubusercontent.com/10797985/52872364-9f377c80-314c-11e9-962a-1fc43b11bf11.png)

package.json
```

{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "eject": "expo eject",
    "test": "node ./node_modules/jest/bin/jest.js --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/samples": "2.1.1",
    "expo": "^32.0.0",
    "expokit": "^32.0.6",
    "react": "16.5.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz",
    "react-native-camera": "^1.11.1",
    "react-native-elements": "^1.0.0",
    "react-native-webview-quilljs": "^0.8.38",
    "react-navigation": "^3.0.9"
  },
  "devDependencies": {
    "babel-preset-expo": "^5.0.0",
    "jest-expo": "^32.0.0"
  },
  "private": true
}


```

Implementation:

```
import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    Platform
} from 'react-native';
import {WebViewQuillEditor, WebViewQuillViewer} from 'react-native-webview-quilljs'

export default class ProfileScreen extends React.Component {
  static navigationOptions = {
    title: 'Profile',
  };


  contentToDisplay = '';
  webViewQuillViewer;
  constructor(props) {
    super(props);
    this.state = {
      messageDelta: '<p>Test</p>'
    };
  }

  onLoadCallback = (e) => { 
    console.log("loaded...", e);
  }

  getDeltaCallback = () => {

  }


  componentWillMount(){
    console.log("mounted");
  }

  render() {
    return (
        <View style={styles.container}>
         <WebViewQuillViewer
                ref={component => (this.webViewQuillViewer = component)}
                contentToDisplay={this.state.messageDelta}
                onLoad={this.onLoadCallback}
          />
        </View>
    )
  }



}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#ffffff',
    paddingTop: 40
  },
  richText: {
    alignItems:'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
});
```

First screen shows what happening on device.

This is log from compliator

```
[18:05:58] Error: invalid asset file path: `${filePath}
[18:05:58]     at Object.parse (/Users/mateusz/projects/mobileTest/node_modules/metro/src/node-haste/lib/AssetPaths.js:66:11)
[18:05:58]     at /Users/mateusz/projects/mobileTest/node_modules/metro/src/Assets.js:312:34
[18:05:58]     at Generator.next (<anonymous>)
[18:05:58]     at step (/Users/mateusz/projects/mobileTest/node_modules/metro/src/Assets.js:326:347)
[18:05:58]     at /Users/mateusz/projects/mobileTest/node_modules/metro/src/Assets.js:326:577
[18:05:58]     at new Promise (<anonymous>)
[18:05:58]     at /Users/mateusz/projects/mobileTest/node_modules/metro/src/Assets.js:326:258
[18:05:58]     at getAsset (/Users/mateusz/projects/mobileTest/node_modules/metro/src/Assets.js:326:58)
[18:05:58]     at /Users/mateusz/projects/mobileTest/node_modules/metro/src/Server.js:576:1722
[18:05:58]     at Generator.next (<anonymous>)
```

I assume that something is wrong with Asset method to require some html.

Please provide any help if you can.

Thanks in advance

**Comment by reggie3** [comment]
I see you have a dependency for expokit.  I built this module for vanilla expo applications, and never tested this module with expokit.

**Comment by mrnetforge** [comment]
I just created plain new project with Expo, and that error still appears. Im not sure that problem is in your code, but every time when i try to use Asset function from Expo that load any of kind html this error appears.

Can i ask you to just init some blank project with Expo and try to install your plugin and give me some feedback?

Thanks in advance

**Comment by morenoh149** [comment]
@mrnetforge maybe share your vanilla project on github and we can clone and run it

**Comment by mrnetforge** [comment]
https://github.com/mrnetforge/react-native-webview-quilljs

Try run this on iOS simulator.

**Comment by morenoh149** [comment]
I don't follow the example, you are trying to render and empty string, and the assets dir has two images. I thought you would want to render an html file?

**Comment by mrnetforge** [comment]
Oh i see.. `contentToDisplay` should contains special format, but anyway, i just want to start editor with empty content and even that:
```
  render() {
    return (
      <View style={styles.container}>
        <Text>Editor Test</Text>
        <WebViewQuillEditor
            ref={component => (this.webViewQuillEditor = component)}
          />
      </View>
    );
  }
}
```



Still error..

**Comment by lorenzsell** [comment]
I am getting the same error message and I'm providing the editor with the delta copy pasted from your example:

```
editorMessageDelta: [
        { insert: 'Hello World' },
        { insert: '!', attributes: { bold: true } }
      ],
```

**Comment by lorenzsell** [comment]
Here is my implementation:

```
state = {
    editorMessageDelta: [
        { insert: 'Hello World' },
        { insert: '!', attributes: { bold: true } }
      ],
  };

  getDeltaCallback = response => {
    console.log('getDeltaCallback');
    console.log(response.delta);
  };

  onLoadCallback = () => {
    console.log('onLoadCallback');
  };


  renderTextInput() {

    console.log('@@@ editorMessageDelta: ', this.state.editorMessageDelta);

    return (
      <View>
        <WebViewQuillEditor
          ref={component => (this.webViewQuillEditor = component)}
          getDeltaCallback={this.getDeltaCallback}
          onLoad={this.onLoadCallback}
        />
      </View>
    );
  }
```

I'm getting the same error as @mrnetforge:

```
[18:19:05] Error: invalid asset file path: `${filePath}
[18:19:05]     at Object.parse (/Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/node-haste/lib/AssetPaths.js:66:11)
[18:19:05]     at /Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Assets.js:312:34
[18:19:05]     at Generator.next (<anonymous>)
[18:19:05]     at step (/Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Assets.js:326:347)
[18:19:05]     at /Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Assets.js:326:577
[18:19:05]     at new Promise (<anonymous>)
[18:19:05]     at /Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Assets.js:326:258
[18:19:05]     at getAsset (/Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Assets.js:326:58)
[18:19:05]     at /Users/lorenzsell/DEV/sutra-app/node_modules/metro/src/Server.js:576:1722
[18:19:05]     at Generator.next (<anonymous>)
```

Any ideas what might be causing this?  Thank you.

**Comment by mrnetforge** [comment]
I need to mention that i tried also another react-native plugin, and i had exactly sam error.

Its must be something with `Asset` class from Expo that required html and inject inside webView component.

Maybe Expo version is the case, in this project package.json dependencies we see:

`"expo": "^30.0.0",`

and in devDependencies

`"expo": "^25.0.0",`

Im my case im using latest expo and expo-cli version.

Tomorrow i will try to downgrade and test that.

**Comment by lorenzsell** [comment]
I'm on expo 32.  I tried adjusting the project package.json dependencies, but still get the error.

**Comment by mrnetforge** [comment]
Yes... same on my side

**Comment by cagigas** [comment]
I'm stuck in the same, any luck?

**Comment by lorenzsell** [comment]
No, I ended up going with a different WYSIWYG editor but the solution I applied might be relevant to you.  I setup a custom webpage on my server optimized to run in a webview in react native.  With enough tweaking, I ended up with a pretty seamless experience that felt native and was relatively performant.

**Comment by morenoh149** [comment]
@lorenzsell what wysiwyg did you use?

**Comment by lorenzsell** [comment]
Froala.

**Comment by morenoh149** [comment]
@lorenzsell did you evaluate https://quilljs.com ?

**Comment by reggie3** [comment]
I just published a new version 0.8.39 which hopefully addresses this issue.  It also removes the library's dependency on Expo.

**Comment by reggie3** [comment]
Fixed in version 2
